var/spack/repos/builtin/packages/python/package.py

```diff
@@ -475,21 +485,46 @@ def write_easy_install_pth(self, exts):
                         "sys.__egginsert = p+len(new)\n")
 
     def activate(self, ext_pkg, **args):
+        path_view = args.get("path_view", "")
+
         ignore = self.python_ignore(ext_pkg, args)
         args.update(ignore=ignore)
 
         super(Python, self).activate(ext_pkg, **args)
 
-        exts = spack.store.layout.extension_map(self.spec)
+        # if we are activating in a filesystem view, we need to remove
+        # the link to the main easy_install.pth file prior to writing
+        # a custom one
+        if path_view:
+            link_pth = join_path(path_view, self.easy_install_file)
+            try:
+                os.remove(link_pth)
+            except OSError as e:
+                if e.errno != errno.ENOENT:  # No such file or directory
+                    raise
+
+        exts = spack.store.layout.extension_map(self.spec, path_view=path_view)
         exts[ext_pkg.name] = ext_pkg.spec
-        self.write_easy_install_pth(exts)
+
+        # also include globally activated extensions in easy_install.pth
+        if path_view:
+            # the directory layout assures that there are no conflicts between
+            # globally and locally activated extensions
+            exts_global = spack.store.layout.extension_map(self.spec,
+                                                           path_view="")
+            # since extension_map() returns a copy, we can update directly
+            exts.update(exts_global)
```

Comment:
> What is the correct behavior in the case of creating a filesystem views while globally activated extensions are already present? Because the extensions' files have already been linked to the python-package's install directory.

I prefer your first approach (fail if anything is globally activated). I think people are going to be attracted to one method over the other. If people get excited about using both then a different PR could add a phase that temporarily deactivates all extensions.